lib/cagen.py

```diff
@@ -186,18 +186,19 @@ def voms(self, vo_name):
         _write_file(lsc, '%s\n%s\n' % (self.host_subject, self.subject))
 
         vomses = '/etc/vomses'
-        contents = '"%s" "%s" "15001 "%s" "%s"\n' % (vo_name, uri, self.host_subject, vo_name)
+        new_vo = '"%s" "%s" "15001" "%s" "%s"\n' % (vo_name, uri, self.host_subject, vo_name)
         try:
             with open(vomses, 'r') as vomses_file:
-                vos = vomses_file.read().rstrip()
-                if contents.strip() in vos:
+                vos = vomses_file.read()
+                if new_vo in vos:
                     return
+                vos += new_vo
```

Comment:
Should check if there's a newline at the end of the file and add it if missing before adding the new vo.